pkg/ingress/query.go

```diff
@@ -52,6 +54,38 @@ func (i *QueryIngress) Get() *networkingv1.Ingress {
 
 	if i.jaeger.Spec.Ingress.IngressClassName != nil {
 		spec.IngressClassName = i.jaeger.Spec.Ingress.IngressClassName
+	} else {
+		class := ""
+		nginxIngressAvailable := false
+		config, err := rest.InClusterConfig()
+		if err == nil {
+			clientSet, err := kubernetes.NewForConfig(config)
+			if err == nil {
+				ingressList, err := clientSet.NetworkingV1().IngressClasses().List(context.Background(), metav1.ListOptions{})
+				if err == nil {
+					for _, ingress := range ingressList.Items {
+						if ingress.Name == "nginx" {
+							nginxIngressAvailable = true
+						}
+						for k, v := range ingress.Annotations {
+							if k == "ingressclass.kubernetes.io/is-default-class" {
+								if v == "true" {
+									class = ingress.Name
+									break
+								}
+							}
+						}
+					}
+				}
+			}
+		}
```

Comment:
Please, instead of this, create a new method and call it from here. It will help readability. Also, instead of checking there are no errors, check if there is an error and return if `true`.